### PR TITLE
feat: added Login Page

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Added Login page and implemented Authentication

# More details:

1. Login Page -
- When a user clicks on `Play now` button, he is redirected to the `Login` page instead. User after he/she logins, is then directed to the rules page.
- User can also use dummy username and password and play as a guest without signing in.

2. 404 Page - 
- Added 404 Page not found. User will be shown this message if she/he types the wrong URL in the browser address bar.

3. Categories - 
- Added some more data in categories file. Now, user has more choices and more quizzes to play.

4. Errors - 
- Fixed some errors I had been facing earlier in the Result page.

5. Responsiveness - 
- Site has been made responsive.

# Screenshots:

loginquiz

### CODE REVIEW

**Review by Asuraking1n — APPROVED** [review]
Features and plaining of project is literally awesome, it exactly feels like I am giving an exam
One thing even after login, login button is visible instead of logout.
And your design could use some padding In mobile view.

**Review by sakshi006 — APPROVED** [review]
Looks good to merge.
One suggestion (optional) : when user is asked to enter the name before the quiz begins, if the input field is empty, you can display an error msg saying something like : please enter your name.

File: src/index.js [hunk 1]
```diff
@@ -9,11 +9,13 @@ import Result from "./Pages/Result/Result";
 import Rules from "./Pages/Rules/Rules";
 import NavBar from "./Components/Navbar/Navbar";
 import Footer from "./Components/Footer/Footer";
+import Login from "./Pages/Login/Login";
 import { CategoryContextProvider } from "./Context/category-context";
```
Comment (ranitasahaa1230) [inline src/index.js:13]: Why are you importing all these files over here? In your pages folder make an index.js file and export all these and then import all these files wherever they required

**Review by ranitasahaa1230 — APPROVED** [review]
Suggested some changes. Looks good to merge

**Comment by krgarima** [comment]
> Features and plaining of project is literally awesome, it exactly feels like I am giving an exam One thing even after login, login button is visible instead of logout. And your design could use some padding In mobile view.

Done button change.

**Comment by krgarima** [comment]
> Looks good to merge. One suggestion (optional) : when user is asked to enter the name before the quiz begins, if the input field is empty, you can display an error msg saying something like : please enter your name.

I tried doing that but since the button is disabled you cannot click on the button and thus the error is not getting set to true. If you have a better way, I would gladly do that.

**Pull request merged**